File: src/main/java/com/team3/gudit/global/exception/ErrorResponse.java
package com.team3.gudit.global.exception;

import java.util.Map;

public record ErrorResponse(
        String code,
        String message,
        Map<String, String> fieldErrors
) {
    // 기본
    public static ErrorResponse from(ErrorCode errorCode) {
        return new ErrorResponse(
                errorCode.getCode(),
                errorCode.getMessage(),
                Map.of()
        );
    }

    // 검증 오류 시
    public static ErrorResponse validation(String code, String message, Map<String, String> fieldErrors) {
        return new ErrorResponse(
                code,
                message,
                fieldErrors
        );
    }

    // 서버 에러
    public static ErrorResponse internalServerError() {
        return new ErrorResponse(
                "COMMON_500",
                "서버 내부 오류가 발생했습니다.",
                Map.of()
        );
    }
}

File: src/main/kotlin/com/team3/gudit/global/exception/ErrorResponse.kt
package com.team3.gudit.global.exception

@JvmRecord
data class ErrorResponse(
    val code: String,
    val message: String,
    val fieldErrors: Map<String, String>,
) {
    companion object {
        @JvmStatic
        fun from(errorCode: ErrorCode): ErrorResponse = ErrorResponse(
            errorCode.getCode(),
            errorCode.getMessage(),
            emptyMap(),
        )

        @JvmStatic
        fun validation(code: String, message: String, fieldErrors: Map<String, String>): ErrorResponse =
            ErrorResponse(code, message, fieldErrors)

        @JvmStatic
        fun internalServerError(): ErrorResponse = ErrorResponse(
            "COMMON_500",
            "서버 내부 오류가 발생했습니다.",
            emptyMap(),
        )
    }
}
